### PR TITLE
Refactoring: components related to request new password feature + flow

### PR DESCRIPTION
**Pull request by cedricmessiant**
I still want to add some unit tests for these components but it is not the priority of the sprint so we should merge this asap

### CODE REVIEW

File: assembl/static2/js/app/components/login/sendPwdForm.jsx [hunk 1]
```diff
@@ -30,14 +34,21 @@ class SendPwdForm extends React.Component {
...
+  };
 
-  submitHandler(e) {
+  handleSubmit = (e: SyntheticEvent<HTMLFormElement>) => {
+    const { requestPasswordChange } = this.props;
+    const slug = getDiscussionSlug();
     e.preventDefault();
-    this.props.sendRequest(this.state.identifier, getDiscussionSlug());
-  }
+    if (this.state.identifier && slug) {
```
Comment (amensouissi) [inline assembl/static2/js/app/components/login/sendPwdForm.jsx]: const {identifier} = this.state

File: assembl/static2/js/app/components/login/sendPwdForm.jsx [hunk 2]
```diff
@@ -1,25 +1,29 @@
...
 
-  componentWillReceiveProps(nextProps) {
-    const { auth } = nextProps;
-    const resp = auth.passwordChangeRequest;
-    if (resp && resp.success === false) {
-      const firstError = resp.data[0];
+class RequestNewPasswordForm extends React.Component<Props, State> {
+  state = { identifier: null };
+
+  componentWillReceiveProps(nextProps: Props) {
```
Comment (amensouissi) [inline assembl/static2/js/app/components/login/sendPwdForm.jsx]: componentWillReceiveProps? https://reactjs.org/docs/react-component.html#unsafe_componentwillreceiveprops

File: assembl/static2/js/app/pages/requestPasswordChange.jsx [hunk 3]
```diff
@@ -1,23 +1,34 @@
...
 import { connect } from 'react-redux';
 import { Grid, Row, Col } from 'react-bootstrap';
+
+import { requestPasswordChangeAction } from '../actions/authenticationActions';
 import SendPwdForm from '../components/login/sendPwdForm';
 import SendPwdConfirm from '../components/login/sendPwdConfirm';
 
-class RequestPasswordChange extends React.Component {
+export type ChangePasswordResponse = {
+  data: Array<any>,
```
Comment (amensouissi) [inline assembl/static2/js/app/pages/requestPasswordChange.jsx]: any ?

File: assembl/static2/js/app/pages/requestPasswordChange.jsx [hunk 4]
```diff
@@ -26,7 +37,11 @@ class RequestPasswordChange extends React.Component {
 }
 
 const mapStateToProps = state => ({
```
Comment (amensouissi) [inline assembl/static2/js/app/pages/requestPasswordChange.jsx]: {auth}

File: assembl/static2/js/app/components/login/sendPwdForm.jsx (hunk 2 above)
Comment (cedricmessiant) [reply assembl/static2/js/app/components/login/sendPwdForm.jsx]: I will not change this in this PR

**Comment by cedricmessiant** [comment]
I added some tests, I will merge now

**Pull request merged**